src/utils/manifest.ts

```diff
@@ -98,3 +117,41 @@ export function convertPatternToRegExp(pattern: string): RegExp {
     ? new RegExp(globToRegExp(normalize(pattern)).toString().slice(1, -2))
     : new RegExp(`^${normalize(pattern)}`);
 }
+
+/**
+ * Determines if the manifest contains the entry at the given relative path.
+ *
+ * @param manifestEntries manifest entries to search
+ * @param entryRelativePathToLookup a relative path to look up in the manifest
+ * @returns `true` if the manifest contains the entry at the given relative path
+ */
+export function containsEntryInManifest(
+  manifestEntries: Record<string, ManifestEntry>,
+  entryRelativePathToLookup: string,
+): boolean {
+  for (const [entryName, entry] of Object.entries(manifestEntries)) {
+    switch (entry.kind) {
+      case "file":
+      case "symlink": {
+        if (entryName === entryRelativePathToLookup) {
+          return true;
+        }
+        break;
+      }
+      case "directory": {
+        if (!entryRelativePathToLookup.startsWith(entryName)) {
+          break;
+        }
+
+        const relativePath = entryRelativePathToLookup.slice(
+          entryName.length + 1,
+        );
+        return containsEntryInManifest(entry.entries, relativePath);
+      }
+      default:
+        unreachable();
```

Comment:
note to self: replace this with `assertNever` once this is released
https://github.com/denoland/std/pull/5690

Reply:
It's now released